android/java/org/chromium/chrome/browser/brave_news/BraveNewsAdapterFeedCard.java

```diff
@@ -0,0 +1,144 @@
+import org.chromium.brave_news.mojom.Deal;
+import org.chromium.brave_news.mojom.FeedItem;
+import org.chromium.brave_news.mojom.FeedItemMetadata;
+import org.chromium.brave_news.mojom.PromotedArticle;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.brave_news.BraveNewsControllerFactory;
+import org.chromium.chrome.browser.brave_news.BraveNewsUtils;
+import org.chromium.chrome.browser.brave_news.models.FeedItemCard;
+import org.chromium.chrome.browser.brave_news.models.FeedItemsCard;
+import org.chromium.chrome.browser.ntp_background_images.util.NTPUtil;
+import org.chromium.mojo.bindings.ConnectionErrorHandler;
+import org.chromium.mojo.system.MojoException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class BraveNewsAdapterFeedCard extends RecyclerView.Adapter<BraveNewsAdapterFeedCard.ViewHolder> {
+
+    private LayoutInflater mInflater;
+    private int mType;
+    private Context mContext;
+    private Activity mActivity;
+    private View view;
+    private LinearLayout linearLayout;
+    private LinearLayout.LayoutParams linearLayoutParams;
+
+    public CopyOnWriteArrayList<FeedItemsCard> mNewsItems;
+    private FeedItemsCard mNewsItem;
+    private CopyOnWriteArrayList<FeedItemCard> mNewsItemsCard;
+	private ViewHolder mHolder;
+    private final String TAG = "BN";
+
+    private BraveNewsController mBraveNewsController;
+
+    public BraveNewsAdapterFeedCard(Activity activity,
+            CopyOnWriteArrayList<FeedItemsCard> newsItems,
+            BraveNewsController braveNewsController) {
+        this.mInflater = LayoutInflater.from(activity);
+        this.mActivity = activity;
+        this.mNewsItems = newsItems;
+        this.mBraveNewsController = braveNewsController;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        view = mInflater.inflate(R.layout.brave_news_row, parent, false);
+        mHolder = new ViewHolder(view);
+
+        return mHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull BraveNewsAdapterFeedCard.ViewHolder holder, int position) {
+        LinearLayout.LayoutParams params1;
+        if (mNewsItems != null){
+            mNewsItem = mNewsItems.get(position);
+            BraveNewsUtils.logFeedItem(mNewsItems.get(position), "createfeed BraveNewsAdapterFeedCard onBindViewHolder position");
+            try {
+                if (mBraveNewsController != null) {
+                    new CardBuilderFeedCard(mBraveNewsController, mHolder.linearLayout, mActivity,
+                            position, mNewsItems.get(position),
+                            mNewsItems.get(position).getCardType());
+                }
+                
+            } catch (Exception e) {
+                Log.e("bn", "crashinvestigation onBindViewHolder e: " + e);
```

Comment:
If you have 
`private final String TAG = "BN";`
maybe better change it to
`Log.e(TAG,...`

Reply:
changed to TAG

Reply:
all logs are using TAG member variable, resolving